lib/cli-engine.js

```diff
@@ -256,6 +256,32 @@ function isErrorMessage(message) {
     return message.severity === 2;
 }
 
+
+/**
+ * Checks if a provided path is a directory and returns a glob string matching
+ * all files under that directory if so, the path itself otherwise.
+ *
+ * Reason for this is that `glob` needs `/**` to collect all the files under a
+ * directory where as our previous implementation without `glob` simply walked
+ * a directory that is passed. So this is to maintain backwards compatibility.
+ *
+ * Also makes sure all path separators are POSIX style for `glob` compatibility.
+ *
+ * @param {string} pathname The directory path to be modified
+ * @returns {string} The glob path or the file path itself
+ */
+function processPath(pathname) {
+    try {
+        var fileStat = fs.statSync(pathname);
+
+        var newPath = fileStat.isDirectory() ? path.join(pathname, "**") : pathname;
+        return newPath.replace(/\\/g, "/");
+    } catch (err) {
+        return pathname;
```

Comment:
This seems like an abuse of `try-catch`. If it's possible that `pathname` isn't a string, it would be better to check it explicitly and also to mention in the JSDoc comment.

Also, your comment mentions `file`, but I don't see a variable with that name.